### PR TITLE
Implementation of CredentialManagement wrapping feature

### PR DESCRIPTION
**Pull request by TMSCH**
List of changes:

- Cleaned up exposed OpenYolo API so that return value of Promise resolution are non null. Errors should be used instead.
- Updated logic to respect the cleaned up API.
- Removed the `wrapBrowserRequest` logic from the initialization. Instead using a per request logic that can be triggered with a `browserWrappingRequired` error.
- Refactored errors to not subclass the JS `Error` class as the prototype is not extendable and it can lead to unexpected behavior in compiled code.
- Creation of a no-op `navigator.credentials` wrapper for browser without it.
- Added better tests coverage for api.ts file.
- Some syntax cleanup, made `navigator_credentials.ts` file to use `await` construct instead of `Promise` syntax.
- Expose a subset of `OpenYoloInternalError` on `openyolo_spi` namespace so that providers' implementations' can return specific errors (`noCredentialsAvailable`, `userCanceled`, `browserWrappingRequired`)
- I had to manually write a subset of the CredentialManagement API externs as I couldn't find any in our own codebase or externally...

Please review very carefully. I tested the code with Smart Lock implementation with success.

### CODE REVIEW

**Comment by TMSCH** [comment]
Tests coverage are >90%! :D

File: ts/api/api_test.ts [hunk 1]
```diff
@@ -288,4 +258,173 @@ describe('OpenYolo API', () => {
...
+      let navCredentialsSpy: jasmine.Spy;
+
+      beforeEach(() => {
+        dispatchSpy = spyOn(operationRequest, 'dispatch');
+        navCredentialsSpy = spyOn(NavigatorCredentials.prototype, methodName);
+      });
+
+      it('dispatches the request', (done) => {
+        dispatchSpy.and.returnValue(Promise.resolve(expectedResult));
+        (openYoloApiImpl[methodName] as methodSignatures[M])(
```
Comment (tikurahul) [inline ts/api/api_test.ts]: Out of curiosity, why do you have to cast the `openYoloApiImpl[methodName] as methodSignatures[M]` ?

Comment (TMSCH) [inline ts/api/api_test.ts]: I think it complained about untyped method, as the argument `methodName` would not give typing information on the method called, but the template would.

**Comment by TMSCH** [comment]
Thanks Rahul!

**Comment by TMSCH** [comment]
Thanks Iain!

**Pull request merged**